mycroft/client/speech/mic.py

```diff
@@ -283,7 +284,7 @@ def decrease_noise(level):
 
     @staticmethod
     def sec_to_bytes(sec, source):
-        return sec * source.SAMPLE_RATE * source.SAMPLE_WIDTH
+        return int(sec * source.SAMPLE_RATE * source.SAMPLE_WIDTH)
```

Comment:
This needs to be rounded up for it to work consistently

either
`return int(math.ceil(sec * source.SAMPLE_RATE * source.SAMPLE_WIDTH))`
or the c hackers choice
`return int(sec * source.SAMPLE_RATE * source.SAMPLE_WIDTH + 0.5)`

Reply:
Hmm, this is strange because it indicates that changing a value by one sample (1/16000 of second) is making a difference. Could you by chance find which call to sec_to_bytes caused that?

Reply:
The relevant call is at L366 in mic.py 

I think it's if the resulting byte size is odd, and since we pull the data from the rear of the byte array every 16 bit sample will be mangled.

Reply:
which means my suggested fix is probably wrong instead we need to check that `bytes % sample_width == 0` and then append `sample_width - bytes % sample_width` padding if it's not?